python/lib/database.py

```diff
@@ -0,0 +1,608 @@
+        sql_print(sql)
+        cursor.execute(sql)
+
+    db.commit()
+
+
+def drop_schemata_except(db):
+    keep_schemata = [
+        "public",
+        "tiger",
+        "tiger_data",
+        "topology",
+        "information_schema",
+        "pg_catalog",
+        "pg_toast",
+        "periods",
+        "__pg_ivm__",
+    ]
+
+    with db.cursor() as cursor:
+        sql = "SELECT schema_name FROM information_schema.schemata;"
+        cursor.execute(sql)
+        all_schemata = [row["schema_name"] for row in cursor.fetchall()]
+
+        for schema in all_schemata:
+            if schema not in keep_schemata:
+                sql = f"DROP SCHEMA IF EXISTS {schema} CASCADE;"
+                sql_print(sql)
+                cursor.execute(sql)
+
+    db.commit()
+
+
+def drop_public_entities(db):
+    tables = ["atd_txdot_locations", "units", "crash_location_map"]
+    views = ["crashes"]
+    materialized_views = ["road_types", "unit_types"]
+
+    with db.cursor() as cursor:
+        for view in views:
+            sql = f"DROP VIEW IF EXISTS public.{view};"
+            sql_print(sql)
+            cursor.execute(sql)
+
+        for materialized_view in materialized_views:
+            sql = f"DROP MATERIALIZED VIEW IF EXISTS public.{materialized_view};"
+            sql_print(sql)
+            cursor.execute(sql)
+
+        for table in tables:
+            sql = f"DROP TABLE IF EXISTS public.{table};"
+            sql_print(sql)
+            cursor.execute(sql)
+
+    db.commit()
+
+
+def pull_down_locations(db):
+    # Run the pg_dump command and pipe the output into psql
+    command = (
+        f"pg_dump -t atd_txdot_locations | PGPASSWORD=vz psql -U vz -h db -d visionzero"
+    )
+    bash_print(command)
+    process = subprocess.Popen(
+        command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    _, error = process.communicate()
+
+    if process.returncode != -1:
+        print(f"Error occurred while running pg_dump and psql: {error}")
+
+
+def create_schemata(db):
+    schemata = [
+        "cris_facts",
+        "visionzero_facts",
+        "cris_lookup",
+        "visionzero_lookup",
+    ]
+
+    with db.cursor() as cursor:
+        for schema in schemata:
+            sql = f"CREATE SCHEMA IF NOT EXISTS {schema};"
+            sql_print(sql)
+            cursor.execute(sql)
+
+    db.commit()
+
+
+def create_lookup_tables(db):
+    sql_commands = [
+        """
+        CREATE SEQUENCE cris_lookup.road_types_sequence;
+        """,
+        """
+        CREATE TABLE cris_lookup.road_types (
+            id INTEGER PRIMARY KEY DEFAULT nextval('cris_lookup.road_types_sequence'),
+            upstream_id INTEGER,
+            description TEXT
+        );
+        """,
+        """
+        CREATE TABLE visionzero_lookup.road_types (
+            id INTEGER PRIMARY KEY DEFAULT nextval('cris_lookup.road_types_sequence'),
+            upstream_id INTEGER,
```

Comment:
I noticed that, when a new VZ lookup is added, this field is `NULL`. Is there ever a case the upstream ID would not be null?